lib/node.js

```diff
@@ -990,23 +990,17 @@ declare module "string_decoder" {
   // TODO
 }
 
-declare class tls$TLSSocket {
-  constructor(socket: typeof net$Socket, options?: Object): void;
-  address(): net$Socket$address;
+declare class tls$TLSSocket extends net$Socket {
+  constructor(socket: net$Socket, options?: Object): void;
   authorized: boolean;
-  authorizationError?: string;
-  static encrypted: boolean;
-  getCiphier(): Object;
-  getEphemeralKeyInfo(): Object;
-  getPeerCertificate(detailed?: boolean): Object;
-  getSession(): Buffer;
-  getTLSTicket(): Buffer;
-  localAddress: string;
-  localPort: number;
-  remoteAddress: string;
-  remoteFamily: string;
-  remotePort: number;
-  renegotiate(options: Object, callback: Function): void;
+  authorizationError: string | null;
+  encrypted: true;
+  getCipher(): { name: string, version: string } | null;
+  getEphemeralKeyInfo(): { type: 'DH', size: number } | { type: 'EDHC', name: string, size: number } | null;
```

Comment:
@avikchaudhuri having `null` here should still work with sentinel props, right? you'd have to do something like

```
if (keyInfo !== null && keyInfo.type === 'EDHC') {
  // I can access keyInfo.name here
}
```